openwillis/measures/text/speech_attribute.py

```diff
@@ -104,58 +106,33 @@ def filter_transcribe(json_conf, measures, speaker_label=None):
         only the relevant data for processing.
     text_list: list
         List of transcribed text.
-         split into words, phrases, turns, and full text.
+         split into words, turns, and full text.
     text_indices: list
         List of indices for text_list.
-         for phrases and turns.
-    Raises:
-    ...........
-    ValueError: If the speaker label is not found in the json response object.
     ------------------------------------------------------------------------------------------------------
     """
     item_data = json_conf["results"]["items"]
-
-    # make a dictionary to map old indices to new indices
-    item_data = cutil.create_index_column(item_data, measures)
+    
+    for i, item in enumerate(item_data): # create_index_column
+        item[measures["old_index"]] = i
 
     # extract text
-    text = " ".join(
-        [
-            item["alternatives"][0]["content"]
-            for item in item_data
-            if "alternatives" in item
-        ]
-    )
-
-    # phrase-split
-    phrases, phrases_idxs = cutil.phrase_split(text)
-
-    # turn-split
-    turns = []
-    turns_idxs = []
+    text = " ".join([item["alternatives"][0]["content"] for item in item_data if "alternatives" in item])
 
     if speaker_label is not None:
+        turns_idxs, turns = cutil.filter_speaker_aws(item_data, min_turn_length, speaker_label)
+    else:
+        turns_idxs, turns = [], []
 
-        turns_idxs, turns, phrases_idxs, phrases = cutil.filter_speaker(
-            item_data, speaker_label, turns_idxs, turns, phrases_idxs, phrases
-        )
-
-    # entire transcript - by joining all the phrases
-    text = " ".join(phrases)
-
-    # filter json to only include items with start_time and end_time
-    filter_json = cutil.filter_json_transcribe(item_data, speaker_label, measures)
-
-    # extract words
+    text = " ".join(turns)
```

Comment:
This should be in the if statement:

i.e. only run if `speaker_label is not None`, o/w keep as previously